nameservice/errors.go

```diff
@@ -26,8 +26,25 @@ var (
 	ErrNotFound        = errors.New("not found")
```

Comment:
It would best if your PR removed the previous errors
```
ErrUnableToResolve = errors.New("unable to resolve")
	ErrNotFound        = errors.New("not found")
	ErrInvalidName     = errors.New("invalid name")
	ErrInvalidAddress  = errors.New("invalid address")
```
Instead the new errors you created are used, see other notes as to where changes would be made to use the new errors. 
One additional change that would need to be made is in `ens/errors.go`
```
func wrapError(err error) error {
	if err == nil {
		return nil
	}
	switch fmt.Sprintf("%v", errors.Cause(err)) {
	case fmt.Sprintf("%v", errors.Cause(errors.Errorf("no resolver"))),
		fmt.Sprintf("%v", errors.Cause(errors.Errorf("No resolution"))):
		return errors.WithMessage(err, nameservice.ErrUnableToResolve.Error())

	case fmt.Sprintf("%v", errors.Cause(errors.Errorf("unregistered name"))):
		return errors.WithMessage(err, nameservice.ErrNotFound.Error())

	case fmt.Sprintf("%v", errors.Cause(errors.Errorf("could not parse address"))):
		// address related to not being able to part ens address not ethereum address
		return errors.WithMessage(err, nameservice.ErrInvalidName.Error())

	default:
		return err
	}
}
```
Based on the switch statement, I'd suggest per case these errors`ErrUnableToResolve`, `ErrInvalidName`, `ErrInvalidAddress` you can remove `IsNoResolverError`, `IsNotFoundError`, `IsInvalidNameError`, `IsInvalidAddressError`

This way ENS, will resolve an unknown `err` e.g. default condition, or a known error one of the case statements. The calling code only needs to use your `IsRfc1035Error` function to determine this.

Reply:
To clarify the above comment as I have made a mistake explaining which errors should be used to replace the old ones. Regarding mapping of errors, I'd suggest something like
```
func wrapError(err error) error {
	if err == nil {
		return nil
	}
	switch fmt.Sprintf("%v", errors.Cause(err)) {
	case fmt.Sprintf("%v", errors.Cause(errors.Errorf("no resolver"))),
		fmt.Sprintf("%v", errors.Cause(errors.Errorf("No resolution"))):
		return errors.WithMessage(err, nameservice.ErrNXDomain.Error())

	case fmt.Sprintf("%v", errors.Cause(errors.Errorf("unregistered name"))):
		return errors.WithMessage(err, nameservice.ErrNXDomain.Error())

	case fmt.Sprintf("%v", errors.Cause(errors.Errorf("could not parse address"))):
		// address related to not being able to part ens address not ethereum address
		return errors.WithMessage(err, nameservice.ErrFormat.Error())

	default:
		return err
	}
}
```